src/systems/physics/Physics.cc

```diff
@@ -727,7 +781,16 @@ void PhysicsPrivate::CreatePhysicsEntities(const EntityComponentManager &_ecm)
         }
         else
         {
-          collisionPtrPhys = linkPtrPhys->ConstructCollision(collision);
+          auto linkCollisionFeature = entityCast(_parent->Data(), linkPtrPhys,
+              this->entityLinkCollisionMap);
+          if (!linkCollisionFeature)
+          {
+            ignwarn << "Can't process Collision entity, physics engine "
+                    << "missing ConstructSdfCollision feature" << std::endl;
+            return true;
```

Comment:
should we still warn about it if we made this feature optional? same with other optional features' warnings.

Reply:
The idea is that the user will only see these warnings if they're trying to use an unsupported feature. If they loaded a world that needs a certain feature, but that feature is not supported, then maybe the user is doing something they don't mean to. For example, a user loads a model that contains joints, but the engine is going to ignore joints. Then we warn them that the joints are being ignored.

But it's true that many times the user may be aware that the feature is not supported and they're fine ignoring it. In this case the warnings will be spammy. For example, the user wants to load the same model into TPE and DART at different times to test different things. They know that the model's joints will be ignored by TPE and they're fine with it. And they don't want to maintain 2 different models, one for TPE and another for DART.

For this situation, how about offering the user a way to silence the warnings? This could be done within the SDF, for example:

```
    <plugin
      filename="libignition-gazebo-physics-system.so"
      name="ignition::gazebo::systems::Physics">
      <engine>
        <filename>ignition-physics-tpesim-plugin</filename>
        <!-- Don't emit warnings if the following unsupported features are used -->
        <suppress_warnings>GetBasicJointState</suppress_warnings>
      </engine>
    </plugin>
```

Reply:
I guess the warnings were helpful when I was debugging TPE. Giving option to suppress it sounds like a good idea, but I think listing feature  in sdf file requires a bit much work and knowledge about features used by each world, which isn't super obvious. How about using `igndbg` instead of `ignwarn`? `igndbg` seems to only show up at a more verbose level if I remember correctly, so that users can still access this info if they notice something wrong.

Reply:
> How about using igndbg instead of ignwarn?

My immediate impulse was to treat it as a warning because I see it as a ["potentially harmful situation"](https://stackify.com/logging-levels-101/) when it isn't straight up an error ("problem that needs to be dealt with"). But given the conversation above and the fact that this could become spammy, I took your suggestion, and made it so each message is only printed once: fa29326 .